pkg/app/app_fetch.go

```diff
@@ -43,6 +44,31 @@ func (a *App) fetch(dstPath string) (string, exec.CmdRunResult) {
 	}
 
 	result = a.runVendir(confReader, dstPath)
+	// retry if error occurs before reporting failure.
+	// This is mainly done to support private registry
+	// authentication for images/bundles since placeholder
+	// secrets may not be populated in time.
+	if result.Error != nil && a.HasImageOrImgpkgBundle() {
```

Comment:
idea: what if we record whether we created any placeholder secrets earlier and only do retry if we did.

Reply:
Would it be enough to have the App check that it has secrets in the particular format of the placeholder secret (i.e. `%s-fetch-%d`)? Otherwise, would this information be on the App status or somewhere else?

Reply:
Added to cleanup list in #336